index.js

```diff
@@ -56,85 +71,142 @@ export type Options = {
+    options
+  );
 }
 
 /**
- * Inquire if the type of local authentication policy (LAPolicy) is supported
- * on this device with the device settings the user chose.
- * @param {object} options LAPolicy option, iOS only
- * @return {Promise} Resolves to `true` when supported, otherwise `false`
+ * Fetches login combination for `service`.
+ * @param {object} options An Keychain options object.
+ * @return {Promise} Resolves to `{ service, username, password, storage }` when successful
  */
-export function canImplyAuthentication(options?: Options): Promise<boolean> {
-  if (!RNKeychainManager.canCheckAuthentication) {
-    return Promise.resolve(false);
-  }
-  return RNKeychainManager.canCheckAuthentication(options);
+export function getGenericPassword(
+  options?: Options
+): Promise<false | SharedWebCredentials> {
+  var obj: { service?: string } = options || {};
+
+  return RNKeychainManager.getGenericPasswordForOptions(obj.service, options);
 }
 
 /**
- * Get what type of hardware biometry support the device has.
- * @return {Promise} Resolves to a `BIOMETRY_TYPE` when supported, otherwise `null`
+ * Deletes all generic password keychain entries for `service`.
+ * @param {object} options An Keychain options object.
+ * @return {Promise} Resolves to `true` when successful
  */
-export function getSupportedBiometryType(): Promise<?($Values<typeof BIOMETRY_TYPE>)> {
-  if (!RNKeychainManager.getSupportedBiometryType) {
-    return Promise.resolve(null);
-  }
-  return RNKeychainManager.getSupportedBiometryType();
+export function resetGenericPassword(options?: Options): Promise<boolean> {
+  var obj: { service?: string } = options || {};
+
+  return RNKeychainManager.resetGenericPasswordForOptions(obj.service, options);
+}
+
+/**
+ * Checks if we have a login combination for `server`.
+ * @param {string} server URL to server.
+ * @param {object} options An Keychain options object.
+ * @return {Promise} Resolves to `{service, storage}` when successful
+ */
+export function hasInternetCredentials(
+  server: string,
+  options?: Options
+): Promise<false | Result> {
+  return RNKeychainManager.hasInternetCredentialsForServer(server, options);
 }
 
 /**
  * Saves the `username` and `password` combination for `server`.
  * @param {string} server URL to server.
  * @param {string} username Associated username or e-mail to be saved.
  * @param {string} password Associated password to be saved.
- * @param {object} options Keychain options, iOS only
- * @return {Promise} Resolves to `true` when successful
+ * @param {object} options An Keychain options object.
+ * @return {Promise} Resolves to `{service, storage}` when successful
  */
 export function setInternetCredentials(
   server: string,
   username: string,
   password: string,
   options?: Options
-): Promise<void> {
+): Promise<false | Result> {
   return RNKeychainManager.setInternetCredentialsForServer(
     server,
     username,
     password,
-    getMinimumSecurityLevel(options),
```

Comment:
This is also a breaking change for the ios native module, since this is not used at all ios side, we can remove `withSecurityLevel:(__unused NSString *)level` on the native method.

https://github.com/oblador/react-native-keychain/blob/31af59a1fdfca1790bd7e77ff0bf9bae61674628/RNKeychainManager/RNKeychainManager.m#L362-L374

Reply:
fixed!